Translation of Apollo Official Chinese Document(s) (#4322)
apolloconfig/apollo: docs/en/faq/common-issues-in-deployment-and-development-phase.md

> Note 2: If you are adding an environment to Apollo Configuration Center that has been running for a while, don't forget to refer to [2.1.2.4 Importing ApolloConfigDB project data from another environment](en/deployment/distributed-deployment-guide#_2124-Importing-ApolloConfigDB-project-data-from-another-environment) to initialize the new environment

#### 4.2 1.5.1 and earlier
##### 4.2.1 Adding an Apollo pre-defined environment

If the environment to be added is an Apollo pre-defined environment (DEV, FAT, UAT, PRO), two steps are required.
1. protaldb add environment, refer to [3.1 Adjusting ApolloPortalDB configuration](en/deployment/distributed-deployment-guide#_31-Adjusting-ApolloPortalDB-configuration)

> 2. add the meta server address corresponding to the new environment for apollo-portal, refer to: [2.2.1.1.2.4 Configuring meta service information for apollo-portal](en/deployment/distributed-deployment-guide#_221124-Configuring-apollo-portal's-meta-service-information). apollo-client also needs to be configured properly when used in a new environment, refer to: [1.2.2 Apollo Meta Server](en/usage/java-sdk-user-guide#_122-Apollo-Meta-Server).

Reviewer: ```suggestion
2. add the meta server address corresponding to the new environment for apollo-portal, refer to: [2.2.1.1.2.4 Configuring meta service information for apollo-portal](en/deployment/distributed-deployment-guide#_221124-configuring-apollo-portal39s-meta-service-information). apollo-client also needs to be configured properly when used in a new environment, refer to: [1.2.2 Apollo Meta Server](en/usage/java-sdk-user-guide#_122-Apollo-Meta-Server).
```

> >Note 1: A set of Portal can manage multiple environments, but each environment needs to deploy a set of Config Service, Admin Service and ApolloConfigDB independently, please refer to: [2.1.2 Creating ApolloConfigDB](en/deployment/distributed-deployment-guide#_212-Creating-ApolloConfigDB), [3.2 Adjusting ApolloConfigDB configuration](en/deployment/distributed-deployment-guide?id=_32-), [3.1.2 Creating ApolloconfigDB](en/deployment/distributed-deployment-guide#_312-apollo.portal.meta.servers---List-of-Meta-Service-for-each-environment) Adjusting apolloconfigdb configuration), [2.2.1.1.2 Configuring database connection information](en/deployment/distributed-deployment-guide#_22112-Configuring-database-connection-information)

Reviewer: ```suggestion
>Note 1: A set of Portal can manage multiple environments, but each environment needs to deploy a set of Config Service, Admin Service and ApolloConfigDB independently, please refer to: [2.1.2 Creating ApolloConfigDB](en/deployment/distributed-deployment-guide#_212-Creating-ApolloConfigDB), [3.2 Adjusting ApolloConfigDB configuration](en/deployment/distributed-deployment-guide?id=_32-), [3.1.2 Creating ApolloconfigDB](en/deployment/distributed-deployment-guide#_312-apolloportalmetaservers-list-of-meta-service-for-each-environment) Adjusting apolloconfigdb configuration), [2.2.1.1.2 Configuring database connection information](en/deployment/distributed-deployment-guide#_22112-Configuring-database-connection-information)
```

    }
    switch (envName.trim().toUpperCase()) {
      ...
      case "BETA":
        return Env.BETA;
      ...
      default:
        return Env.UNKNOWN;
    }
  }
}
```
4. Modify [apollo-env.properties](https://github.com/apolloconfig/apollo/blob/master/apollo-portal/src/main/resources/apollo-env.properties) to add the ``beta.meta`` placeholder.
```properties
local.meta=http://localhost:8080
dev.meta=${dev_meta}
fat.meta=${fat_meta}
beta.meta=${beta_meta}
uat.meta=${uat_meta}
lpt.meta=${lpt_meta}
pro.meta=${pro_meta}
```
5. Modify the [com.ctrip.framework.apollo.core.internals.LegacyMetaServerProvider](https://github.com/apolloconfig/apollo/blob/master/apollo-core/src/main/java/com/ctrip/framework/apollo/core/internals/LegacyMetaServerProvider.java) class to add logic to read the meta server address of the ``BETA`` environment.
```java
public class LegacyMetaServerProvider {
    ...
    domains.put(Env.BETA, getMetaServerAddress(prop, "beta_meta", "beta.meta"));
    ...
}
```

6. Add `BETA` environment for protaldb, refer to [3.1 Adjusting ApolloPortalDB configuration](en/deployment/distributed-deployment-guide#_31-Adjusting-ApolloPortalDB-configuration)

> 7. Add the meta server address corresponding to the new environment for apollo-portal, refer to: [2.2.1.1.2.4 Configuring meta service information for apollo-portal](en/deployment/distributed-deployment-guide#_221124-Configuring-apollo-portal's-meta-service-information). apollo-client also needs to be configured properly when used in a new environment, refer to: [1.2.2 Apollo Meta Server](en/usage/java-sdk-user-guide#_122-Apollo-Meta-Server).

Reviewer: ```suggestion
7. Add the meta server address corresponding to the new environment for apollo-portal, refer to: [2.2.1.1.2.4 Configuring meta service information for apollo-portal](en/deployment/distributed-deployment-guide#_221124-configuring-apollo-portal39s-meta-service-information). apollo-client also needs to be configured properly when used in a new environment, refer to: [1.2.2 Apollo Meta Server](en/usage/java-sdk-user-guide#_122-Apollo-Meta-Server).
```

> >Note 1: A set of Portal can manage multiple environments, but each environment needs to deploy a set of Config Service, Admin Service and ApolloConfigDB independently, please refer to: [2.1.2 Creating ApolloConfigDB](en/deployment/distributed-deployment-guide#_212-Creating-ApolloConfigDB), [3.2 Adjusting ApolloConfigDB configuration](en/deployment/distributed-deployment-guide?id=_32-), [3.1.2 Creating ApolloconfigDB](en/deployment/distributed-deployment-guide#_312-apollo.portal.meta.servers---List-of-Meta-Service-for-each-environment) Adjusting apolloconfigdb configuration), [2.2.1.1.2 Configuring database connection information](en/deployment/distributed-deployment-guide#_22112-Configuring-database-connection-information)

Reviewer: ```suggestion
>Note 1: A set of Portal can manage multiple environments, but each environment needs to deploy a set of Config Service, Admin Service and ApolloConfigDB independently, please refer to: [2.1.2 Creating ApolloConfigDB](en/deployment/distributed-deployment-guide#_212-Creating-ApolloConfigDB), [3.2 Adjusting ApolloConfigDB configuration](en/deployment/distributed-deployment-guide?id=_32-adjusting-apolloconfigdb-configuration), [3.1.2 Creating ApolloconfigDB](en/deployment/distributed-deployment-guide#_312-apolloportalmetaservers-list-of-meta-service-for-each-environment) Adjusting apolloconfigdb configuration), [2.2.1.1.2 Configuring database connection information](en/deployment/distributed-deployment-guide#_22112-Configuring-database-connection-information)
```

@Configuration
@EnableTransactionManagement
@PropertySource(value = {"classpath:configservice.properties"})
@ComponentScan(basePackageClasses = {ApolloCommonConfig.class,
    ApolloBizConfig.class,
    ConfigServiceApplication.class,
    ApolloMetaServiceConfig.class})
public class ConfigServiceApplication {
  ...
}
```

#### 2. Modify `eureka.service.url` in ApolloConfigDB.ServerConfig table to point to your own Eureka address

For example, if your own Eureka service address is 1.1.1.1:8761 and 2.2.2.2:8761, then set eureka.service.url in ApolloConfigDB.ServerConfig table to

```
http://1.1.1.1:8761/eureka/,http://2.2.2.2:8761/eureka/
```

Note that changing the Eureka address only requires changing the `eureka.service.url` in the ApolloConfigDB.ServerConfig table, not the meta server address.

> By default, the meta service and config service are deployed in the same JVM process, so the address of the meta service is the address of the config service, so you don't need to change the meta server address when you modify the Eureka address.

### 9. Using `ConditionalOnProperty` in Spring Boot does not read the configuration

> The `@ConditionalOnProperty` feature is supported since version 0.10.0, see [Spring Boot integration method](en/usage/java-sdk-user-guide?id=_3213-spring-boot integration method (recommended))

Reviewer: ```suggestion
The `@ConditionalOnProperty` feature is supported since version 0.10.0, see [Spring Boot integration method](en/usage/java-sdk-user-guide?id=_3213-spring-boot-integration-methods-recommended)
```
